src/vs/editor/contrib/stickyScroll/browser/stickyScrollController.ts

```diff
@@ -3,63 +3,281 @@
 			this._onContextMenu(event);
 		}));
+
+		// Context key which indicates if the sticky scroll is focused or not
+		this._stickyScrollFocusedContextKey = EditorContextKeys.stickyScrollFocused.bindTo(this._contextKeyService);
+		// Create a focus tracker to track the focus on the sticky scroll widget
+		const focusTracker = this._register(dom.trackFocus(this._stickyScrollWidget.getDomNode()));
+		this._register(focusTracker.onDidBlur(_ => {
+			this._disposeFocusStickyScrollStore();
+		}));
+
+		// Register the click link gesture to allow the user to click on the sticky scroll widget to go to the definition
+		this._register(this._createClickLinkGesture());
 	}
 
-	get stickyScrollCandidateProvider() {
+	get stickyScrollCandidateProvider(): IStickyLineCandidateProvider {
 		return this._stickyLineCandidateProvider;
 	}
 
-	get stickyScrollWidgetState() {
+	get stickyScrollWidgetState(): StickyScrollWidgetState {
 		return this._widgetState;
 	}
 
+	// Get function to allow actions to use the sticky scroll controller
+	public static get(editor: ICodeEditor): IStickyScrollController | null {
+		return editor.getContribution<StickyScrollController>(StickyScrollController.ID);
+	}
+
+	private _disposeFocusStickyScrollStore() {
+		this._focused = false;
+		this._stickyScrollFocusedContextKey.set(false);
+		this._focusedStickyElement?.classList.remove('focus');
+		this._stickyScrollWidget.getDomNode().blur();
+		this._focusDisposableStore!.dispose();
+	}
+
+	public focus(): void {
+		this._focused = true;
+		const focusState = this._stickyScrollFocusedContextKey.get();
+		const rootNode = this._stickyScrollWidget.getDomNode();
+		this._stickyElements = rootNode.children;
+		this._numberStickyElements = this._stickyElements.length;
+
+		if (focusState === true || this._numberStickyElements === 0) {
+			// If lready focused or no line to focus on, return
+			return;
+		}
+		this._focusDisposableStore = new DisposableStore();
+		this._stickyScrollFocusedContextKey.set(true);
+		this._focusedStickyElement = rootNode.lastElementChild! as HTMLDivElement;
+		this._focusedStickyElement.classList.add('focus');
+		// The first line focused is the bottom most line
+		this._focusedStickyElementIndex = this._numberStickyElements - 1;
+
+		// Whenever the mouse hovers on the sticky scroll remove the keyboard focus
+		this._focusDisposableStore.add(dom.addDisposableListener(rootNode, dom.EventType.MOUSE_OVER, () => {
+			this._focusedStickyElement?.classList.remove('focus');
+		}));
+		// Whenever the mouse leaves the sticky scroll, set the keyboard focus to be on the last line hovered
+		this._focusDisposableStore.add(dom.addDisposableListener(rootNode, dom.EventType.MOUSE_OUT, () => {
+			this._focusedStickyElement = this._stickyScrollWidget.lastMouseFocusedStickyLine;
+			this._focusedStickyElementIndex = this._stickyScrollWidget.lastMouseFocusedStickyLineIndex;
+			this._focusedStickyElement?.classList.add('focus');
+		}));
```

Comment:
Careful with that: moving the mouse shouldn't change the focus, only the hover feedback but the focused element should stay the same.

Reply:
Ah okay, the reason I wrote it this way, is because the previous usage of the click link gesture made it so that, when you hover onto the sticky widget, the lines are in a way `focused` and the background color becomes grey. Were this piece of code removed, there would be two `focused` or greyed-out lines, one from the click link gesture and one from the actual keyboard focus.

Reply:
But I'll refactor the code so that the focus() function is used in the click link gesture.